test/Parse/diagnose_availability.swift

```diff
@@ -0,0 +1,27 @@
+// RUN: %target-typecheck-verify-swift 
+
+// SR-4231: Misleading/wrong error message for malformed @available
+
+@available(OSX 10.6, *) // no error
+func availableSince10_6() {}
+
+@available(OSX, introduced: 10.0, deprecated: 10.12) // no error
+func introducedFollowedByDeprecated() {}
+
+@available(OSX 10.0, deprecated: 10.12)
```

Comment:
For visibility (got folded by the force-pushes): Can you make this recover?  There's a decl down there, but this second diagnostic makes it seems like there isn't.